src/linker/configuration.py

```diff
@@ -42,20 +44,20 @@ class Config:
 
     def __init__(
         self,
-        pipeline_specification: Path,
-        input_data: Path,
-        computing_environment: Optional[Path],
+        pipeline_specification: Union[str, Path],
+        input_data: Union[str, Path],
+        computing_environment: Optional[Union[str, Path]],
+        results_dir: Union[str, Path],
     ):
         # Handle pipeline specification
-        self.pipeline_specification_path = pipeline_specification
```

Comment:
in my opinion, we shouldn't keep these paths around in the `config` object after initialization; anything needed should be loaded into the dictionary and/or saved into the results dir. So I removed these attributes and made the  `_copy_configuration_files_to_results_directory` method take on appropriate parameters.

Reply:
The reason I did it this way is because I didn't want to copy the files over until Pipeline() got validated as well. That said, I don't care that strongly and this is certainly cleaner. It's not like we're copying huge files or anything

Reply:
Ah, gotcha. That makes sense. I think a good middle ground is to:
--pass the paths into `main()`
--instantiate the Config there
--keep the copy fn as a static method
--only call it after the pipeline validates

Then we only copy after the pipeline validates, but don't store the paths in the Config.